### ISSUE TITLE
With "tf" dimension ordering no learning improvement

### ISSUE THREAD

**Issue by Trickery94** [fchollet/keras#4243]
I noticed that when using the "tf" dimension ordering, the training loss and accuracy do not improve at all. For example, using the [keras cifar 10 example](https://github.com/fchollet/keras/blob/master/examples/cifar10_cnn.py) if you add `set_image_dim_ordering("tf")` at the top, the accuracy stays at 0.1. The training data however has the correct ordering of (32, 32, 3).

I tried it with Keras 1.1.1, Tensorflow 0.11.0rc1, Cuda 8.0 and Keras 1.1.1, Tensorflow 0.10.0, Cuda 7.5

Edit: 
I found out that it is caused by the optimizer. For example, SGD and Adagrad result in no accuracy improvement, but when using adadelta it works just fine with "tf" ordering. 
However, it is still only limited to the cifar example, in the mnist example "tf" ordering works just fine with any optimizer.

Edit2:
I noticed that sometimes the accuracy actually improves with "tf" ordering. It seems to be random behaviour, which may result from the weight initializations (I use the default "glorot_uniform"). In around 1 of 5 runs the accuracy improves beyond 0.1 in the first epoch. With for example "he_normal" it works every time.

**Comment by anewlearner** [comment]
Same problem in keras 1.1.0 with tensorflow 0.10.0.

**Comment by jmccormac** [comment]
I believe this is a valid issue.  There is no need to add `set_image_dim_ordering("tf")` however when using the CIFAR 10 example, if you add `set_image_dim_ordering("th")` the learning improvement is performs as expected.  

Something seems to go wrong only when the image_dim_ordering is `"tf"` in this example, even though this should not have an impact as the training data has been correctly transposed.

**Comment by Trickery94** [comment]
I found out that it is caused by the SGD optimizer. Once you change it to for example adagrad, the accuracy improves even in "tf" ordering mode.

**Comment by hwelsen** [comment]
i'm experiencing the same with theano

with dim_ordering "th" and training data formatted like (data_x, data_y, num_channels)
training Convolutional2D layers either takes ages per epoch (when using a cpu) or yields no learning at all (when using the gpu).
However, when I switch to "tf" and use data in shape of (num_channels, data_x, data_y) it performs as expected.
Is the backend ignoring the dim_ordering?
